File: submissions/0071-simplify-path/solution.ts
function simplifyPath(path: string): string {
  const parts = path.split('/');
  const stack = [];

  for (const part of parts) {
    if (part === '..') {
      stack.pop();
    } else if (part !== '' && part !== '.') {
      stack.push(part);
    }
  }

  return '/' + stack.join('/');
}

File: submissions/0071-simplify-path/solution.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        directories = [""]

        for part in path.split("/"):
            
              if part == "" or part == ".":
                continue
        
              elif part == "..":
                if len(directories) == 1:
                    continue
                else:    
                    directories.pop()
              
              else:
                directories.append(part) 
        
        if len(directories) == 1:
            directories.append("")
        
        return "/".join(directories)
